### PR TITLE
Use shared test facilities for time-tracking and dynamic-delay

### PR DESCRIPTION
**Pull request by oschaaf**
Switch tests for these extensions to use the recent shared test
facilities. Eliminate tests we generalize in #533.

Split out from #512

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

----- 

- [x] #533 needs to land first to avoid trending low on coverage in between.

### CODE REVIEW

**Comment by mum4k** [comment]
@dubious90 please review and assign back to me once done.

File: test/server/http_dynamic_delay_filter_integration_test.cc
Comment (dubious90) [inline test/server/http_dynamic_delay_filter_integration_test.cc:43]: Contxt for a few comments here: If I understand correctly, we're saving some effort by having these all be one test to prevent having to initialize a filter repeatedly (which I assume is slow?).  That makes sense here.

However, I think it would be good if we clarified individual test cases within the TEST_P block a little.
1. "Don't send any config request header" is describing what we are doing - can we add in what behavior we're expecting this to produce?

2.  Can we space out all of the individual test cases separately:

e.g.
```// Comment for test 1
getResponse();
EXPECT_EQ()

// Comment for test 2
...```

File: test/server/http_dynamic_delay_filter_integration_test.cc
Comment (dubious90) [inline test/server/http_dynamic_delay_filter_integration_test.cc:51]: Rather than saying this should become effective, can we describe what the behavior should be?

File: test/server/http_dynamic_delay_filter_integration_test.cc
Comment (dubious90) [inline test/server/http_dynamic_delay_filter_integration_test.cc:54]: Where is this 1600 coming from?

File: test/server/http_time_tracking_filter_integration_test.cc
Comment (dubious90) [inline test/server/http_time_tracking_filter_integration_test.cc:64]: same comments here about describing behaviors

**Review by dubious90 — COMMENTED** [review]
migration looks good.  Also have one chage-level question:
I'm surprised there are no corresponding bazel changes - do we need to bring in a dependency on the new fixture and remove the old one?

**Comment by oschaaf** [comment]
@dubious90 feedback applied in 2fa83ee326b7ce5792e29074bfe8b5fac28a60c4
The build file changes have already been merged in https://github.com/envoyproxy/nighthawk/pull/517/files, which is why they no longer show up here.

**Review by dubious90 — APPROVED** [review]

**Comment by oschaaf** [comment]
With #540 merged, the merge of this one needs to hold off until #533 lands, to avoid a coverage dip because of the redundant test elimination that we perform here. Adding that as a task in the PR description.

**Review by mum4k — APPROVED** [review]
Looks good.

Can you kick Circle CI to ensure all checks pass?

**Comment by oschaaf** [comment]
> Looks good.
> 
> Can you kick Circle CI to ensure all checks pass?

Done, merged master in here, and tests are passing now.

**Pull request merged**
